### DISCUSSION TITLE
Starter Guide 👩‍💻 & FAQs 👀

### DISCUSSION THREAD

**Discussion by Praful932** [Praful932/Kitabe#42] [General]
Hello everyone 👋! Hope you're doing well. Thank you for showing interest 😍 in our project. 
In this discussion post I'll go through a detailed step-by-step guide to quickly get started with contributions.
If you have any doubt in any step feel free to comment down below, We'll try our best to answer your question😊

1. First step is to Understand what the Project is about 🧐, go through the [Readme](https://github.com/Praful932/Kitabe) and try to get an overall idea of what the Project is about & do check out the [video ](https://www.youtube.com/watch?v=N2179jIbijw&ab_channel=Projects)demo too if not already. Try the project out live by going to the [website](https://kitabe-app.herokuapp.com/).
2. Now that you have a solid understanding 😇 of the project, try setting up the project locally. Follow the **Setting Up the Project guide** [here](https://github.com/Praful932/Kitabe/blob/master/CONTRIBUTING.md#setting-up-the-project). See if you are able to run the project locally on your device.
3. Awesome! So far so good 👌. Next step is browse the existing [unassigned](https://github.com/Praful932/Kitabe/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee) issues, See if there are any issues that might interest you 🙂. If you find one, great, proceed to next step. 
If not browse the [project](https://kitabe-app.herokuapp.com), see if you can find any bug that's annoying  or any additional feature you wish there could be in this project.Before you go ahead and create one, do check the [existing issues](https://github.com/Praful932/Kitabe/issues) to know whether there are already any assigned issues or you will end up creating one. Once you know your issue isn't there, go to the [issues](https://github.com/Praful932/Kitabe/issues/new/choose) section and create an issue with appropriate details or if you want to have a discussion with us feel free to create one [here](https://github.com/Praful932/Kitabe/discussions/new).
4. Now that you have found/created an appropriate issue that you feel like you can contribute, just follow the **Contributing** steps [here](https://github.com/Praful932/Kitabe/blob/master/CONTRIBUTING.md#contributing) right from getting assigned to an issue and making a Pull Request for the same. As  simple as that :) 

**Very Important Points to be Noted 🔴**
- You have to **strictly follow Issue template and Pull request template** when creating either of them. They are automatically rendered according to the type of Issue/PR and you just have to fill in the blanks. For Fields that do not apply just mention N/A. **No Issue/PR will be considered which doesn't follow the format**.
- For Issues that Involve UI/UX Changes, First a wireframe/mockup design has to be created to show, after approval/changes suggested by maintainers, code change can be done.
- You can be **only** assigned to **one issue at a time**, unless you submit a Pull request for the same and it is merged or you are unassigned you cannot be assigned to any other issue.
- If you are **assigned and unable to contribute**, please mention that in the issue so others who want to contribute can take up that issue.
- If you are assigned, It is expected that you **make a Pull Request at the earliest** even if it's incomplete or WIP, so we can review to know you are going in the right direction.
- When branching, always branch out from `master` branch.

**FAQs**
1. **How to keep my fork copy updated?**
Ans - Sometimes, what happens is that, you forked your project copy yesterday and today I made a commit to the master branch, which isn't reflected in your forked copy's master branch. To keep the fork updated, use this [website](https://upriver.github.io/).

We'll keep updating the **FAQs** as Questions arise.

Feel free to drop any questions in the discussions.

Happy Contributing :)

**Comment by Praful932** [comment]
Tagging all current contributors @micksharma @Anurag-bhatia @padmajabhol @machine-sg @saumyaparikh @subhangi2731 @saaalik
@sk1122 @srimontidutta @codebuzzer01 @Shivam-Shandilya1 @abhish3k-11 @vermatanu @srimontidutta @Mukta-Sawant @ranjal007 @supzi-del @samfubuki @Vartika511 @sahildotexe
to take a look 👀

cc @maneprajakta

**Comment by Greeshma2903** [comment]
I have used Upriver to update my forked copy, now to view changes locally on pc, should I use:
`git pull origin master` ...?

@Praful932

**Reply by Praful932** [reply]
Yes
